src/tests.py

```diff
@@ -236,13 +238,127 @@ def _integration_test(target_directory):
             assert str(e) == "Branch does not have a parent: master"
 
 
+
+def _test_clean_branches(target_directory):
+    # type: (Text) -> None
+    target_directory = os.path.expanduser(target_directory)
+    target_container = os.path.dirname(target_directory)
+    assert not os.path.exists(target_directory)
+    assert os.path.isdir(target_container)
+
+    with run_test(target_directory):
+        # Initialize a repo and add a first commit so we can tell what branch we're on.
+        print("Initializing repo")
+        git("init")
+        open("hello.txt", "w").close()
+        git("add .")
+        git("commit -am initial_commit")
+
+        assert get_current_branch() == "master"
+        original_commit = hash_for("HEAD")
+
+        # Create all the branches
+        make_child_branch("valid_branch")
+        assert get_current_branch() == "valid_branch"
+
+        git("checkout master")
+
+        make_child_branch("ghost_branch_childless")
+        assert get_current_branch() == "ghost_branch_childless"
+
+        git("checkout master")
+
+        make_child_branch("ghost_branch_with_children")
+        assert get_current_branch() == "ghost_branch_with_children"
+
+        make_child_branch("child_of_ghost_branch")
+        assert get_current_branch() == "child_of_ghost_branch"
+
+        git("checkout master")
+
+        print("Deleting branches from git")
+        git("branch -D ghost_branch_childless")
+        git("branch -D ghost_branch_with_children")
+
+        print("Test cleaning by archiving")
+        clean_invalid_branches(True)
+
+        with get_branch_tracker() as tracker:
+            assert(tracker.is_archived("valid_branch")==False)
+            assert(tracker.is_archived("ghost_branch_childless")==True)
+            assert(tracker.is_archived("ghost_branch_with_children")==True)
+
+            print("Clear archived flags for next test")
+            tracker.set_is_archived("ghost_branch_childless", False)
+            tracker.set_is_archived("ghost_branch_with_children", False)
+
+        print("Test cleaning by deleting")
+        clean_invalid_branches(False)
+
+        with get_branch_tracker() as tracker:
+            assert(tracker.is_branch_tracked("valid_branch")==True)
+            assert(tracker.is_branch_tracked("ghost_branch_childless")==False)
+            assert(tracker.is_branch_tracked("ghost_branch_with_children")==True)
+
```

Comment:
nit: There should be 2 blank lines here instead of 1, according to PEP 8 (eg needs another line)

Reply:
👍